rainmaker_orchestrator/server.py

```
@@ -1,426 +1,714 @@
-                "content": result["content"],
-                "status": "success"
-            }), 200
-        else:
-            return jsonify({
-                "error": result["message"],
-                "status": result["status"]
-            }), 404 if "not found" in result["message"].lower() else 500
+        full_path = Path(settings.workspace_path) / file_path
+
+        if not full_path.exists():
+            raise HTTPException(status_code=404, detail="File not found")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)